index.js

```diff
@@ -58,23 +58,23 @@
 //.
 //. ```javascript
 //. //    env :: Array Type
-//. const env = $.env.concat([List($.Unknown)]);
+//. const env = $.env .concat ([List ($.Unknown)]);
```

Comment:
The space after the method name is to appease the linter. If we want to write `f (x)` the linter wants us to write `foo.bar (baz)` as well. I was experimenting with surrounding method names with spaces to make them appear more like infix operators: `xs .concat (ys)`. What do you think?

Reply:
I see a `:thumbsdown:` from Gabe. I certainly don't love the proposal. Let's vote!

  - :laughing: `$.env.concat([List ($.Unknown)])` ✘
  - :tada: `$.env.concat ([List ($.Unknown)])` ✔︎
  - :heart: `$.env .concat ([List ($.Unknown)])` ✔︎

The ✔︎/✘ indicates whether we'll be able to continue to use [`func-call-spacing`][1].


[1]: https://eslint.org/docs/rules/func-call-spacing

Reply:
I like the idea of the formatting as a signal that curried functions are being used.
```js
[1, 2, 3].concat([4, 5, 6]); // Nothing special going on here
S.concat ([1, 2, 3]) ([4, 5, 6]); // FP in the house!
```

Reply:
I can see the logic behind that, Gabe. What about unary functions, though? `Math.abs(x)` or `Math.abs (x)`? `S.trim(s)` or `S.trim (s)`?

Reply:
What about `require('express')(b, c)`? What about `fs.reduce(f, x)(y)`? I think trying to categorise function calls based on underlying structure will raise more questions than it settles. I vote for the consistency of *always* putting a space between the function or method identifier:

* `require ('express') (a, b)`
* `fs.reduce (f, x) (y)`
* `$.env.concat ([List ($.Unknown)])`

Reply:
`f(x)(y).z` is problematic. In `f (x) (y).z` the `.z` property access appears to be bound to `(y)` rather than to `f (x) (y)`. There are two reasonable solutions, each of which violates one of our current rules:

  - `f (x) (y) .z` (violates [`no-whitespace-before-property`][1])
  - `(f (x) (y)).z` (violates [`no-extra-parens`][2])

I prefer the latter as it is consistent with `foo.bar.baz` (I'm not ready to switch to `foo .bar .baz`).


[1]: https://eslint.org/docs/rules/no-whitespace-before-property
[2]: https://eslint.org/docs/rules/no-extra-parens

Reply:
So with this new approach, all preceding occurrences of `Z.env .concat` can be fixed, right?

Reply:
:zap: